### ISSUE TITLE
What about custom classes?

### ISSUE THREAD

**Issue by loverde-co** [popcornomnom/Extension-FileTemplate#1]
I have my own class and i want to extend it. How to deal with plist to fit like yours but looking for swift file correctly ? Thanks

**Comment by popcornomnom** [comment]
@loverde-co, as I understand correctly, your template should create a one `.swift` file with a custom implementation? So, from the `.plist` file, all you need is the name of the file you are creating?

In this case, your `.plist` contains the file name `___VARIABLE_name:identifier___YourFilePrefixOptional` (by key "Default"), which you can use in the `.swift` file. 
For example:
````
class ___VARIABLE_name___ {
}
````
Of course your `.swift` file can contain any implementation :)

Maybe you can share the code for a better understanding of the problem.

**Comment by loverde-co** [comment]
The problem is the subclass folder, where it (`.plist` file ) look for that folder exist, if doesnt, it (`.swift` file ) was not created. Even if i let it (`.swift` file ) on root folder.
If it is not clear, please, tell me, and i try to explain more :)
Thanks

**Comment by popcornomnom** [comment]
@loverde-co, now I got it :) 
If you also use a `class` type field, the structure of your template folder should be like this:

```
. 📁Template folder
├── TemplateIcon.png
├── TemplateIcon@2x.png
├── TemplateInfo.plist
├── 📁Foo folder
│   └──  ___FILEBASENAME___.swift
└── ...
```
What about `.plist`:

```
.......
<key>Options</key>
<array>
        .......
	<dict>
		<key>Identifier</key>
		<string>subclassName</string>
		<key>Type</key>
		<string>class</string>
		<key>Name</key>
		<string>Subclass of:</string>
		<key>Default</key>
		<string>Foo</string>        <---- This name have to match
						 the name of the folder with 
						the swift file implementation
		<key>Description</key>
		<string>What class to subclass in the new file</string>
		<key>FallbackHeader</key>
		<string>#import &lt;UIKit/UIKit.h&gt;</string>
		<key>Required</key>
		<string>YES</string>
		<key>NotPersisted</key>
		<true/>
	</dict>
	.......
</array>
.......
```

Let me know if this helped :]

**Comment by loverde-co** [comment]
i already did that :) and it not work. I have to user a normal "class and subclass screen" to do what a want. But like your screen ( few fields ), doesnt work

**Comment by popcornomnom** [comment]
@loverde-co can you share your `.plist` for more details?

**Comment by loverde-co** [comment]
Sure! He we go!
[TemplateInfo.plist.zip](https://github.com/popcornomnom/Extension-FileTemplate/files/3951877/TemplateInfo.plist.zip)

**Comment by popcornomnom** [comment]
@loverde-co, good

The name of your folder(s) with the implementation file(s) should be named with a combination of the `Default` values of the elements of the `class`, `combo` or `text` types, or for the `popup` of all `Values' (**the order of the Items matters**).

In your case, **Item0** doesn’t have a value of `Default`, **Item1** is of type `class` with a value of `Default` “UIViewController”, and for the **Item2** you should cover all the `Values` that you listed (these are "Swift" and "Objective-C")

In short, you should create two folders: "UIViewControllerSwift" and "UIViewControllerObjective-C".

**Comment by loverde-co** [comment]
Like i said, i want an extension of my custom subclass with extension on file name. All that configuration i try. And only works with this trick. But i want only one field like yours. But the config of plist doenst fit and not work, because it couldnt find the folder ( anyname ). I Appreciated your help :)

**Comment by popcornomnom** [comment]
Do you want to subclass from custom class from your current project instead of subclassing from UIKit classes?
In my example, you can subclass from any class (if you write it in text field), but the template gives autosuggestions with UIKit classes.

**Comment by loverde-co** [comment]
Yep, with my custom class from current project. Not UIKit.
Thats why i want to force it without using any UIKit.
I notice your examples only work with NSObject. If i try to use another, i need to have the folder that class i want to subclass it.

**Comment by popcornomnom** [comment]
Actually no :) if you write any name in the text field (the disadvantage is that you should write it without any hints), this will create a file with default implementation from `NSObject` folder.
I'm not sure that `plist` allows you to get all classes from the current project into a file template. Of course, you can make a workaround and and list all your own classes manually into `Values`, but I'm not sure if this is what you are looking for.

**Comment by loverde-co** [comment]
No no, this was impossible to make by hand rsrsrsr
My wish is to get a default folder, for any class name i put into text field, and it look for this default folder, name de .swift file ( just like it do now in this plist i show to you ) and do a rename with extension final name ( myclassnameExtension.swift ). Just with one text field like your screen.
